robottelo/hosts.py

```diff
@@ -0,0 +1,368 @@
+
+        # Red Hat Access Insights requires RHEL 6/7 repo and it is not
+        # possible to sync the repo during the tests, Adding repo file.
+        if rhel_distro == DISTRO_RHEL6:
+            rhel_repo = settings.rhel6_repo
+            insights_repo = settings.rhai.insights_client_el6repo
+        if rhel_distro == DISTRO_RHEL7:
+            rhel_repo = settings.rhel7_repo
+            insights_repo = settings.rhai.insights_client_el7repo
+
+        missing_repos = []
+        if insights_repo is None:
+            missing_repos.append('RHAI client')
+        if rhel_repo is None:
+            missing_repos.append('RHEL')
+        if missing_repos:
+            raise ContentHostError(
+                f'Missing {" and ".join(missing_repos)} '
+                f'repository configuration for {rhel_distro}.'
+            )
+
+        self.configure_rhel_repo(rhel_repo)
+
+        self.execute(f'wget -O /etc/yum.repos.d/insights.repo {insights_repo}')
+
+        # Install redhat-access-insights package
+        package_name = 'insights-client'
+        result = self.execute(f'yum install -y {package_name}')
+        if not result.status:
+            raise ContentHostError('Unable to install redhat-access-insights package')
+
+        # Verify if package is installed by query it
+        result = self.execute(f'rpm -qi {package_name}')
+        logger.info(f'Insights client rpm version: {result.stdout}')
+        if not result.status:
+            raise ContentHostError('Unable to install redhat-access-insights package')
+
+        # Register client with Red Hat Access Insights
+        result = self.execute('insights-client --register')
+        if not result.status:
+            raise ContentHostError(
+                'Unable to register client to Access Insights through Satellite'
+            )
+
+    def set_infrastructure_type(self, infrastructure_type="physical"):
+        """Force host to appear as bare-metal or virtual machine in
+        subscription-manager fact.
+
+        :param str infrastructure_type: One of "physical", "virtual"
+        """
+        script_path = "/usr/sbin/virt-what"
+        self.execute(f"cp -n {script_path} {script_path}.old")
+
+        script_content = ["#!/bin/sh -"]
+        if infrastructure_type == "virtual":
+            script_content.append("echo kvm")
+        script_content = "\n".join(script_content)
+        self.execute(f"echo -e '{script_content}' > {script_path}")
+
+    def patch_os_release_version(self, distro=DISTRO_RHEL7):
+        """Patch VM OS release version.
+
+        This is needed by yum package manager to generate the right RH
+        repositories urls.
+        """
+        if distro == DISTRO_RHEL7:
+            rh_product_os_releasever = REPOS['rhel7']['releasever']
+        else:
+            raise ContentHostError('No distro package available to retrieve release version')
+        return self.execute(
+            "touch /etc/yum/vars/releasever "
```

Comment:
You do not need to use `touch` first; just echo "Hello" > kilroy.txt will create the file you need.